src/main/java/org/apache/tools/bzip2/CBZip2InputStream.java

```diff
@@ -29,957 +29,918 @@
-		final byte[] recvDecodingTables_pos = new byte[N_GROUPS]; //        6 byte
-		//---------------
-		//    60798 byte
-		
-		int[] tt; //  3600000 byte
-		byte[] ll8; //   900000 byte
-		
-		//---------------
-		//  4560782 byte
-		//===============
-		
-		Data(int blockSize100k) {
-			super();
-			
-			this.ll8 = new byte[blockSize100k * BZip2Constants.baseBlockSize];
-		}
-		
-		/**
-		 * Initializes the {@link #tt} array.
-		 *
-		 * This method is called when the required length of the array is known.
-		 * I don't initialize it at construction time to avoid unneccessary
-		 * memory allocation when compressing small files.
-		 */
-		final int[] initTT(int length) {
-			int[] ttShadow = this.tt;
-			
-			// tt.length should always be >= length, but theoretically
-			// it can happen, if the compressor mixed small and large
-			// blocks.  Normally only the last block will be smaller
-			// than others.
-			if ((ttShadow == null) || (ttShadow.length < length)) {
-				this.tt = ttShadow = new int[length];
-			}
-			
-			return ttShadow;
-		}
-		
-	}
+                if (nextSym <= 16) {
+                    for (int j = nextSym - 1; j > 0; ) {
+                        yy[j] = yy[--j];
+                    }
+                } else {
+                    System.arraycopy(yy, 0, yy, 1, nextSym - 1);
+                }
+
+                yy[0] = tmp;
+
+                if (groupPos == 0) {
+                    groupPos = G_SIZE - 1;
+                    zt = selector[++groupNo] & 0xff;
+                    base_zt = base[zt];
+                    limit_zt = limit[zt];
+                    perm_zt = perm[zt];
+                    minLens_zt = minLens[zt];
+                } else {
+                    groupPos--;
+                }
+
+                int zn = minLens_zt;
+
+                // Inlined:
+                // int zvec = bsR(zn);
+                while (bsLiveShadow < zn) {
+                    final int thech = inShadow.read();
+                    if (thech >= 0) {
+                        bsBuffShadow = (bsBuffShadow << 8) | thech;
+                        bsLiveShadow += 8;
+                        continue;
+                    } else {
+                        throw new IOException("unexpected end of stream");
+                    }
+                }
+                int zvec = (bsBuffShadow >> (bsLiveShadow - zn)) & ((1 << zn) - 1);
+                bsLiveShadow -= zn;
+
+                while (zvec > limit_zt[zn]) {
+                    zn++;
+                    while (bsLiveShadow < 1) {
+                        final int thech = inShadow.read();
+                        if (thech >= 0) {
+                            bsBuffShadow = (bsBuffShadow << 8) | thech;
+                            bsLiveShadow += 8;
+                            continue;
+                        } else {
+                            throw new IOException("unexpected end of stream");
+                        }
+                    }
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) This block of commented-out lines of code should be removed. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://titan.maxgamer.org:9000/coding_rules#rule_key=squid%3ACommentedOutCodeLine)